core/services/offchainreporting/OCRContract.go

```diff
@@ -0,0 +1,232 @@
+package offchainreporting
+
+import (
+	"context"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/offchainreporting/confighelper"
+	"github.com/smartcontractkit/chainlink/offchainreporting/gethwrappers/offchainaggregator"
+	ocrtypes "github.com/smartcontractkit/chainlink/offchainreporting/types"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/pkg/errors"
+)
+
+var (
+	OCRContractConfigSet = getConfigSetHash()
+)
+
+type (
+	OCRContract struct {
+		ethClient        eth.Client
+		contractFilterer *offchainaggregator.OffchainAggregatorFilterer
+		contractCaller   *offchainaggregator.OffchainAggregatorCaller
+		contractAddress  gethCommon.Address
+		logBroadcaster   eth.LogBroadcaster
+		jobID            models.ID
+		transmitter      Transmitter
+		contractABI      abi.ABI
+		logger           logger.Logger
+	}
+
+	Transmitter interface {
+		CreateEthTransaction(ctx context.Context, toAddress gethCommon.Address, payload []byte) error
+		FromAddress() gethCommon.Address
+	}
+)
+
+var (
+	_ ocrtypes.ContractConfigTracker      = &OCRContract{}
+	_ ocrtypes.ContractTransmitter        = &OCRContract{}
+	_ ocrtypes.ContractConfigSubscription = &OCRContractConfigSubscription{}
+	_ eth.LogListener                     = &OCRContractConfigSubscription{}
+)
+
+func NewOCRContract(address gethCommon.Address, ethClient eth.Client, logBroadcaster eth.LogBroadcaster, jobID models.ID, transmitter Transmitter, logger logger.Logger) (o *OCRContract, err error) {
+	contractFilterer, err := offchainaggregator.NewOffchainAggregatorFilterer(address, ethClient)
+	if err != nil {
+		return o, errors.Wrap(err, "could not instantiate NewOffchainAggregatorFilterer")
+	}
+
+	contractCaller, err := offchainaggregator.NewOffchainAggregatorCaller(address, ethClient)
+	if err != nil {
+		return o, errors.Wrap(err, "could not instantiate NewOffchainAggregatorCaller")
+	}
+
+	contractABI, err := abi.JSON(strings.NewReader(offchainaggregator.OffchainAggregatorABI))
+	if err != nil {
+		return o, errors.Wrap(err, "could not get contract ABI JSON")
+	}
+
+	return &OCRContract{
+		ethClient,
+		contractFilterer,
+		contractCaller,
+		address,
+		logBroadcaster,
+		jobID,
+		transmitter,
+		contractABI,
+		logger,
+	}, nil
+}
+
+func (oc *OCRContract) Transmit(ctx context.Context, report []byte, rs, ss [][32]byte, vs [32]byte) error {
```

Comment:
Is there a better type we could use here? perhaps `[]common.Hash` ?

Reply:
I don't think so

Reply:
What is `rs`, `ss`, `vs` ?

Reply:
ECDSA signatures

Reply:
so perhaps `type Signature [32]byte`

Reply:
The rs, ss, and vs together form the signatures. We'll probably generalize this notion at some point in the future, but for now I'd just leave as is.